policy_definitions/domainjoin_windows/modules/archetypes/lib/policy_assignments/policy_assignment_qby_deploy_windows_domainjoin.tmpl.json

```
@@ -0,0 +1,43 @@
+{
+  "type": "Microsoft.Authorization/policyAssignments",
+  "apiVersion": "2022-06-01",
+  "name": "Deploy-Windows-DomainJoin",
+  "location": "${default_location}",
+  "dependsOn": [],
+  "identity": {
+    "type": "SystemAssigned"
+  },
+  "properties": {
+    "description": "Deploy Windows Domain Join Extension with keyvault configuration when the extension does not exist on a given windows Virtual Machine",
+    "displayName": "Deploy Windows Domain Join Extension with keyvault configuration",
+    "policyDefinitionId": "${root_scope_resource_id}/providers/Microsoft.Authorization/policyDefinitions/Deploy-Windows-DomainJoin",
+    "enforcementMode": "Default",
+    "nonComplianceMessages": [
+      {
+        "message": "Windows Domain Join extension will be deployed on applicable images."
```

```suggestion
        "message": "Windows Domain Join extension is not configured for VM."
```
That is the message shown, why a resource is not compliant. So it should state, what is wrong, not what will happen.